compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -406,7 +406,7 @@ pub fn provide(providers: &mut ty::query::Providers) {
 /// The reason for this separate call is to resolve what would otherwise
 /// be a cycle. Consider this example:
 ///
-/// ```rust
+/// ```ignore HELP (compiler suggests adding another lifetime variable to Sub. why?)
```

Comment:
`Sub` is declared with a lifetime parameter, so it needs to be defined with one as well. Not sure what it should be. @jackh726 may know.

Reply:
I think would be
```rust
trait Sub<'b>: for<'a> Base<'a> {
    type SubItem: for<'a> Sub<'a, BaseItem = &'b u32>;
}
```

Reply:
Or just
```rust
trait Sub<'b>: for<'a> Base<'a> {
    type SubItem: Sub<'b, BaseItem = &'b u32>;
}
```

Not sure, not entirely sure what this is supposed to be doing